File: PSC/Series/Serie-04/Documents/java/MinDistValidator.java

public final class MinDistValidator implements ExamValidator.OneExamValidator {
    private int minDist = 13;
	public void setArgs(String dist) {
		minDist = Integer.parseInt(dist); 
	}
	public String ruleName() { return "Distancia minima entre exames da mesma UC.";	}
	public boolean isValid(Exam exam) { 
		return (exam.date2() - exam.date1()) >= minDist; 
	}
}

File: PSC/Series/Serie-04/src/MinDistValidator.c
#include <stdio.h>
#include 'myLib.h'
#include 'MinDistValidator.h';


static mdvMethods mdv_vtable={
	dtor,
	setArgs,
	ruleName,
	isValid
};

mdv* mdv_ctor(){
	mdv* m = (mdv *)malloc(sizeof(mdv));
	m->vptr = &mdv_table;                     
	return m;
}

static void setArgs(mdv* this, String dest){
	this->minDist = parseInt(dest);
}

static String ruleName(mdv* this){
	return "Distancia minima entre exames da mesma UC.";
}

static boolean isValid(mdv* this, Exam *exam){
	return (exam->vptr->date2() - exam->vptr->date1() >= this->minDist);
}

static void dtor(mdv *this){
	if (this != null)
		free(this);
}



